Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +204,33 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null  || index == 0)
+            {
+                int[] newArray = { 10 };
+                return newArray;
+            }
+
+            if (index < 0 || index > array.Length)
+            {
+                return array;
+            }
+
+            var amendedArray = new int[array.Length + 1];
+
+            for (int i = 0, j = 0; i < array.Length; i++, j++)
+            {
+                amendedArray[i] = array[i];
+                if (j == index)
+                {
+                    j++;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not update the loop counter 'j' within the loop body.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4904050)

Reply:
I can initialise 'j' above for loop, that wouldn't make sense.
also 'j' is not being used as counter 'i' is. This is not an issue.

Reply:
The real issue here is that you use j normally within the for header and then update it once again in the for loop which can be quite confusing. Especially because you do have another if statement afterwards. You should try to make this simpler/more understandable for others.